diff --git a/lib/replace/replace.h b/lib/replace/replace.h
index 051583c..1fa3cb9 100644
--- a/lib/replace/replace.h
+++ b/lib/replace/replace.h
@@ -1122,6 +1122,7 @@ extern const char hexchars_upper[];
 # ifdef HAVE_PTHREAD
 # error Configure failed to detect pthread library with missing TLS support
 # endif
+#define __thread
 #define HAVE___THREAD
 #endif
 
diff --git a/lib/replace/wscript b/lib/replace/wscript
index 4621d90..5d2ed87 100644
--- a/lib/replace/wscript
+++ b/lib/replace/wscript
@@ -703,7 +703,7 @@ syscall(SYS_copy_file_range,0,NULL,0,NULL,0,0);
         msg='Checking for __thread local storage')
 
     if conf.CONFIG_SET('HAVE_PTHREAD') and not conf.CONFIG_SET('HAVE___THREAD'):
-        conf.fatal('Missing required TLS support in pthread library')
+        conf.msg('Missing required TLS support in pthread library', 'continuing; pthread is removed after configure')
 
     conf.CHECK_FUNCS_IN('crypt', 'crypt', checklibc=True)
     conf.CHECK_FUNCS_IN('crypt_r', 'crypt', checklibc=True)
