diff --git a/lib/tevent/tevent_threads.c b/lib/tevent/tevent_threads.c
index 2dc1f17..5cc9e41 100644
--- a/lib/tevent/tevent_threads.c
+++ b/lib/tevent/tevent_threads.c
@@ -394,6 +394,7 @@ static int tevent_threaded_context_destructor(
 		DLIST_REMOVE(main_ev->threaded_contexts, tctx);
 	}
 
+#ifdef HAVE_PTHREAD
 	/*
 	 * We have to coordinate with _tevent_threaded_schedule_immediate's
 	 * unlock of the event_ctx_mutex. We're in the main thread here,
@@ -416,6 +417,7 @@ static int tevent_threaded_context_destructor(
 		abort();
 	}
 
+#endif
 	return 0;
 }
 
