diff --git a/lib/socket/interfaces.c b/lib/socket/interfaces.c
--- a/lib/socket/interfaces.c
+++ b/lib/socket/interfaces.c
@@ -125,6 +125,10 @@ void make_net(struct sockaddr_storage *pss_out,
 	make_bcast_or_net(pss_out, pss_in, nmask, false);
 }
 
+#if defined(__NetBSD__)
+#include "tc_netbsd_iflist.c"
+#endif
+
 #ifdef HAVE_ETHTOOL
 static void query_iface_speed_from_name(const char *name, uint64_t *speed)
 {
@@ -222,6 +226,9 @@ done:
 
 static int _get_interfaces(TALLOC_CTX *mem_ctx, struct iface_struct **pifaces)
 {
+#if defined(__NetBSD__)
+	return tc_netbsd_get_interfaces(mem_ctx, pifaces);
+#else
 	struct iface_struct *ifaces;
 	struct ifaddrs *iflist = NULL;
 	struct ifaddrs *ifptr = NULL;
@@ -336,6 +343,7 @@ static int _get_interfaces(TALLOC_CTX *mem_ctx, struct iface_struct **pifaces)
 
 	*pifaces = ifaces;
 	return total;
+#endif
 }
 
 static int iface_comp(struct iface_struct *i1, struct iface_struct *i2)
