diff --git a/lib/num.h b/lib/num.h
index 423bdf00f..a206dd7b7 100644
--- a/lib/num.h
+++ b/lib/num.h
@@ -26,7 +26,17 @@
 #include "gnutls_int.h"
 
 #include <minmax.h>
-#include <byteswap.h>
+#include <sys/bswap.h>
+
+#ifndef bswap_16
+#define bswap_16 bswap16
+#endif
+#ifndef bswap_32
+#define bswap_32 bswap32
+#endif
+#ifndef bswap_64
+#define bswap_64 bswap64
+#endif
 
 /* data should be at least 3 bytes */
 inline static uint32_t _gnutls_read_uint24(const uint8_t *data)
diff --git a/lib/random.c b/lib/random.c
index b7e90ee2d..eb865258f 100644
--- a/lib/random.c
+++ b/lib/random.c
@@ -51,7 +51,7 @@ static gl_tls_key_t ctx_key;
 
 /* Flag to indicate initialization
  */
-static _Thread_local unsigned rnd_initialized = 0;
+static unsigned rnd_initialized = 0;
 
 static void free_ctx(const void *ctx)
 {
diff --git a/lib/fips.c b/lib/fips.c
index b846cc2ce..cb36786aa 100644
--- a/lib/fips.c
+++ b/lib/fips.c
@@ -57,10 +57,10 @@
 /* We provide a per-thread FIPS-mode so that an application
  * can use gnutls_fips140_set_mode() to override a specific
  * operation on a thread */
 static gnutls_fips_mode_t _global_fips_mode = -1;
-static _Thread_local gnutls_fips_mode_t _tfips_mode = -1;
+static gnutls_fips_mode_t _tfips_mode = -1;
 
-static _Thread_local gnutls_fips140_context_t _tfips_context = NULL;
+static gnutls_fips140_context_t _tfips_context = NULL;
 
 static int _skip_integrity_checks = 0;
 
