diff --git a/lib/fips.c b/lib/fips.c
index b846cc2ce..cb36786aa 100644
--- a/lib/fips.c
+++ b/lib/fips.c
@@ -57,10 +57,10 @@
 /* We provide a per-thread FIPS-mode so that an application
  * can use gnutls_fips140_set_mode() to override a specific
  * operation on a thread */
 static gnutls_fips_mode_t _global_fips_mode = -1;
-static _Thread_local gnutls_fips_mode_t _tfips_mode = -1;
+static gnutls_fips_mode_t _tfips_mode = -1;
 
-static _Thread_local gnutls_fips140_context_t _tfips_context = NULL;
+static gnutls_fips140_context_t _tfips_context = NULL;
 
 static int _skip_integrity_checks = 0;
 
